src/physics/base/Interaction.hh

```diff
@@ -65,7 +71,24 @@ CELER_FUNCTION Interaction Interaction::from_absorption()
 
 //---------------------------------------------------------------------------//
 /*!
- * Whether the interaction succeeded
+ * Construct an interaction for edge cases where there is no state change.
+ */
+CELER_FUNCTION Interaction Interaction::from_unchanged(
+    const units::MevEnergy energy, const Real3 direction)
+{
+    CELER_EXPECT(energy.value() > 0);
+
+    Interaction result;
+    result.action    = Action::unchanged;
+    result.energy    = energy;
+    result.direction = direction;
+    normalize_direction(&result.direction);
```

Comment:
If the interaction is "unchanged" then the track's direction should already be normalized. Can you change to an assertion:
```c++
  CELER_EXPECT(is_soft_unit_vector(direction, SoftEqual<real_type>()));
```
Also the direction should be const reference, and the `energy` doesn't need to be `const`.